### PR TITLE
chore: remove dry run from publish workflow and update documentation

### PR DESCRIPTION
**Pull request by christjt**
#### Type of change

Chore 

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-551

## Description :pencil:

Adds a workflow for automatically publishing Reveal to NPM when a new Github release is made.

Note that if you want to release from an older branch, the workflow must also be present on that branch. Once this is merged I will retrofit this for release/3.2.x branch. For other branches it will not work (unless you also backport the workflow to those branches).

## How has this been tested? :mag:

This has been tested using dry-run. But actual sucess will need to be verified by actual publishing.

## Test instructions :information_source:

:( 

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: RELEASING.md
Comment (haakonflatval-cognite) [inline RELEASING.md:41]: Should we remove this part - or wait until the threejs- peer dependency PR is merged?

File: RELEASING.md
Comment (haakonflatval-cognite) [inline RELEASING.md:39]:
```suggestion
    Use the following template:
```
Triggering my OCD

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

File: RELEASING.md
Comment (christjt) [reply RELEASING.md:41]: Let's wait for peer dependency. We may hit something that ends up messing this up.

**Review by larsmoa — APPROVED** [review]
Major upgrade of the procedure!

**Pull request merged**
